### PR TITLE
chore(Data table): clickable row items

### PR DESCRIPTION
**Pull request by Ratheeshrajan**
Contributes to #1617

 Data table extension: clickable row items data table extension with side panel.

#### What did you change?

- 'onRowClick' trigger the side panel open and show the data of the selected row in the side panel.
- Toggle selected row style
- Indication of the selected row.

#### How did you test and verify your work?

Storybook

### CODE REVIEW

**Review by matthewgallo — CHANGES_REQUESTED** [review]
I don't _think_ that the SidePanel functionality should be embedded inside of the Datagrid. Seems like it would be good to give the consumers of Datagrid the flexibility to implement it how they want (slide in vs slide over side panel, the size of the panel, etc.).

We can keep your implementation as an example, but I think all of the SidePanel handling should be done in Datagrid.stories.js.

File: packages/cloud-cognitive/src/components/Datagrid/Datagrid/DatagridRow.js [hunk 1]
```diff
@@ -23,6 +23,7 @@ const DatagridRow = (datagridState) => {
     <TableRow
       className={cx(`${blockClass}__carbon-row`, {
         [`${blockClass}__carbon-row-expanded`]: row.isExpanded,
+        [`bx--data-table--selected`]: row.isSelected,
```
Comment (matthewgallo) [inline packages/cloud-cognitive/src/components/Datagrid/Datagrid/DatagridRow.js:26]: This line should import the carbon prefix from our settings like the following:

```jsx
import { carbon } from '../../settings';

...
...
[`${carbon.prefix}--data-table--selected`]: row.isSelected,
...
...
```

Comment (Ratheeshrajan) [reply packages/cloud-cognitive/src/components/Datagrid/Datagrid/DatagridRow.js]: @matthewgallo `${carbon.prefix}` class updated.

**Review by matthewgallo — APPROVED** [review]
🎉 looks great @Ratheeshrajan

**Pull request merged**